libfreerdp/core/gateway/arm.c

```diff
@@ -311,6 +318,319 @@ static char* arm_create_request_json(rdpArm* arm)
 	return message;
 }
 
+static WINPR_CIPHER_CTX* treatAuthBlob(const BYTE* pbInput, size_t cbInput)
+{
+	WINPR_CIPHER_CTX* ret = NULL;
+	char algoName[100];
+
+	SSIZE_T algoSz =
+	    ConvertWCharNToUtf8((WCHAR*)pbInput, cbInput / sizeof(WCHAR), algoName, sizeof(algoName));
+	if (algoSz <= 0)
+	{
+		WLog_ERR(TAG, "invalid algoName");
+		return NULL;
+	}
+
+	algoName[algoSz] = 0;
+	if (strcmp(algoName, "AES") != 0)
+	{
+		WLog_ERR(TAG, "only AES is supported for now");
+		return NULL;
+	}
+
+	cbInput -= (algoSz + 1) * sizeof(WCHAR);
+	/* BCRYPT_KEY_DATA_BLOB_HEADER */
+	PBCRYPT_KEY_DATA_BLOB_HEADER header =
```

Comment:
Maybe a good idea to wrap the `pbInput` into a static stream?

Reply:
Hum good point, I'm wondering if the exporting code really cares about the endianess...